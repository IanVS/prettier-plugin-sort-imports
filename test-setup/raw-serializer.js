'use strict';

import { expect } from 'vitest';

const RAW = Symbol.for('raw');

expect.addSnapshotSerializer({
    print(val) {
        return val[RAW];
    },
    test(val) {
        return (
            val &&
            Object.prototype.hasOwnProperty.call(val, RAW) &&
            typeof val[RAW] === 'string'
        );
    },
});
