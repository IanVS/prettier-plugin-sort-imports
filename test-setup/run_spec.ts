'use strict';

import fs from 'fs';
import { extname } from 'path';

import { format } from 'prettier';
import { expect, test } from 'vitest';

import * as plugin from '../src';

export async function run_spec(dirname, parsers, options) {
    options = Object.assign(
        {
            plugins: [plugin],
            tabWidth: 4,
        },
        options,
    );

    /* instabul ignore if */
    if (!parsers || !parsers.length) {
        throw new Error(`No parsers were specified for ${dirname}`);
    }

    for (const filename of fs.readdirSync(dirname)) {
        const path = dirname + '/' + filename;
        if (
            extname(filename) !== '.snap' &&
            fs.lstatSync(path).isFile() &&
            filename[0] !== '.' &&
            filename !== 'ppsi.spec.ts'
        ) {
            const source = read(path).replace(/\r\n/g, '\n');

            const mergedOptions = Object.assign({}, options, {
                parser: parsers[0],
            });
            test(`${filename} - ${mergedOptions.parser}-verify`, async () => {
                const output = await prettyprint(source, path, mergedOptions);
                expect(
                    raw(source + '~'.repeat(80) + '\n' + output),
                ).toMatchSnapshot(filename);
            });

            for (const parserName of parsers.slice(1)) {
                test(`${filename} - ${parserName}-verify`, async () => {
                    const output = await prettyprint(
                        source,
                        path,
                        mergedOptions,
                    );
                    const verifyOptions = Object.assign(mergedOptions, {
                        parser: parserName,
                    });
                    const verifyOutput = await prettyprint(
                        source,
                        path,
                        verifyOptions,
                    );
                    expect(output).toEqual(verifyOutput);
                });
            }
        }
    }
}

async function prettyprint(src, filename, options) {
    return await format(
        src,
        Object.assign(
            {
                filepath: filename,
            },
            options,
        ),
    );
}

function read(filename) {
    return fs.readFileSync(filename, 'utf8');
}

/**
 * Wraps a string in a marker object that is used by `./raw-serializer.js` to
 * directly print that string in a snapshot without escaping all double quotes.
 * Backticks will still be escaped.
 */
function raw(string) {
    if (typeof string !== 'string') {
        throw new Error('Raw snapshots have to be strings.');
    }
    return { [Symbol.for('raw')]: string };
}
