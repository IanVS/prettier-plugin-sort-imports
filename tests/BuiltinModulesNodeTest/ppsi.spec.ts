import {run_spec} from '../../test-setup/run_spec';

run_spec(__dirname, ["typescript"], {
    importOrder: [
    "",
    "<BUILTIN_MODULES>",
    "",
    "<THIRD_PARTY_MODULES>",
    "",
    "^[.]",
    ],
});
