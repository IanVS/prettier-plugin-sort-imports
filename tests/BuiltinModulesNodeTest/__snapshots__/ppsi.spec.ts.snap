// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`groups-node-test-with-builtin-modules.js - typescript-verify > groups-node-test-with-builtin-modules.js 1`] = `
import nodeUtil from 'node:util';

import nodeNodeTestTest from 'node:node:test:test';
import nodeNodeTest from 'node:node:test';
import nodeTestTest from 'node:test:test';
import nodeTest from 'node:test';
import test from 'test'
import node from 'node';

import fs from 'fs';
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
import fs from "fs";
import nodeTest from "node:test";
import nodeUtil from "node:util";

import node from "node";
import nodeNodeTest from "node:node:test";
import nodeNodeTestTest from "node:node:test:test";
import nodeTestTest from "node:test:test";
import test from "test";

`;
